compiler/modules/types/list.bal

```diff
@@ -185,8 +188,8 @@ function listFormulaIsEmpty(Context cx, Conjunction? pos, Conjunction? neg) retu
             return true;
         }
         // Ensure that we can use isNever on rest in listInhabited
-        if cellInner(rest) != NEVER && isEmpty(cx, rest) {
-            rest = cellContaining(cx.env, NEVER);
+        if cellInnerVal(rest) != NEVER && isEmpty(cx, rest) {
```

Comment:
Given that the rest type of a list (at least in cases where it goes via `defineListTypeWrapped`) is a union of "rest type" and UNDEF (whereas previously it was just the "rest type") I am not sure that their intersections can have a rest type that is empty (It must at least have UNDEF). Either way, I think we can get rid of this whole if block since `listInhabited` is no longer dependent on any member being NEVER on the positive side (I think we got rid of that some time ago but never removed this part here).

Reply:
+1 it never gets empty with undef.